dspace-api/src/main/java/org/dspace/util/SolrUpgradeStatistics6.java

```diff
@@ -0,0 +1,713 @@
+                        docs.add(input);
+                        ++numProcessed;
+                        if (numProcessed % batchSize == 0) {
+                                printTime(String.format("\t%,12d Processed...", numProcessed), false);
+                                batchUpdateStats();
+                                if (context.getCacheSize() > CACHE_LIMIT) {
+                                        refreshContext();
+                                }
+                        }
+                }
+                return numProcessed - initNumProcessed;
+        }
+        
+        
+        /*
+         * Map solr fields from legacy ids to UUIDs.
+         * 
+         * The id field is interpreted by the type field.
+         * The scopeId field is interpreted by scopeType field.
+         * 
+         * In order to prevent the re-processing of items with an id that cannot be resolved, the suffix "-legacy" will be appended to any id fields that cannot be resolved.  
+         * 
+         * @param input The SOLR statistics document to be updated
+         * @param col The SOLR field to update (if present)
+         */
+        private void mapField(SolrInputDocument input, FIELD col) throws SQLException {
+                SolrInputField ifield = input.get(col.name());
+                if (ifield != null) {
+                        Collection<Object> vals = ifield.getValues();
+                        ArrayList<String> newvals = new ArrayList<>();
+                        for(Object oval: vals) {
+                                try {
+                                        int legacy = Integer.parseInt(oval.toString());
+                                        UUID uuid = null;
+                                        if (col == FIELD.id) {
+                                                Object otype = input.getFieldValue("type");
+                                                if (otype != null) {
+                                                        int type = Integer.parseInt(otype.toString());
+                                                        uuid = mapType(type, legacy);
+                                                }
+                                        } else if (col == FIELD.scopeId) {
+                                                Object otype = input.getFieldValue("scopeType");
+                                                if (otype != null) {
+                                                        int type = Integer.parseInt(otype.toString());
+                                                        uuid = mapType(type, legacy);
+                                                }
+                                        } else {
+                                                uuid = mapId(col, legacy);
+                                        }
+                                        if (uuid != null) {
+                                                newvals.add(uuid.toString());
```

Comment:
@tomdesair , thank you for the testing and for the feedback!  

As you raise the issue of SOLR schema, it reminds me that I need to confirm that the DSpace UI correctly uses the upgraded statistics.  

- [ ] Validate statistics counts in the UI after the upgrade

If we decide to modify the SOLR schema, it would be useful to have a field such as "dspaceVer" to identify the version compatibility of a statistics record.  This would facilitate subsequent upgrades.  If we implement this, I would leave legacy ids that cannot be mapped unchanged.

@tdonohue , do you have any guidance on updating the SOLR schema in a point release?  

If we update the schema, is there an equivalent SOLR migration that would need to run?

Other TODOs

- [ ] Add space warning to script start

- [ ] Add total duration to status messages